### PR TITLE
Adding ARM template support for Provisioning Services

### PR DESCRIPTION
**Pull request by kvish**
Adding ARM template support under Public preview API version for Provisioning Services.

### CODE REVIEW

**Comment by Tiano2017** [comment]
Some tests are failing. Please fix them.

**Comment by kvish** [comment]
can you share your test output? I am not getting failures locally

File: schemas/2014-04-01-preview/deploymentTemplate.json [hunk 1]
```diff
@@ -216,6 +216,7 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2015-10-31/Microsoft.Automation.json#/resourceDefinitions/jobSchedules" },
                   { "$ref": "https://schema.management.azure.com/schemas/2015-10-01/Microsoft.Media.json#/resourceDefinitions/mediaServices" },
                   { "$ref": "https://schema.management.azure.com/schemas/2016-02-03/Microsoft.Devices.json#/resourceDefinitions/IotHubs" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2017-08-21-preview/Microsoft.Devices.json#/resourceDefinitions/ProvisioningServices" },
```
Comment (TianoMS) [inline schemas/2014-04-01-preview/deploymentTemplate.json]: >                   [](start = 160, length = 18)

nit: remove the extra spaces.

File: schemas/2015-01-01/deploymentTemplate.json [hunk 2]
```diff
@@ -215,6 +215,7 @@
                   { "$ref": "https://schema.management.azure.com/schemas/2015-10-31/Microsoft.Automation.json#/resourceDefinitions/jobSchedules" },
                   { "$ref": "https://schema.management.azure.com/schemas/2015-10-01/Microsoft.Media.json#/resourceDefinitions/mediaServices" },
                   { "$ref": "https://schema.management.azure.com/schemas/2016-02-03/Microsoft.Devices.json#/resourceDefinitions/IotHubs" },
+                  { "$ref": "https://schema.management.azure.com/schemas/2017-08-21-preview/Microsoft.Devices.json#/resourceDefinitions/ProvisioningServices" },
```
Comment (TianoMS) [inline schemas/2015-01-01/deploymentTemplate.json]: >                   [](start = 160, length = 18)

here as well.

**Pull request merged**